images/ceph/toolbox.sh

```diff
@@ -30,7 +30,7 @@ write_endpoints() {
     endpoints=$(cat ${MON_CONFIG})
 
     # filter out the mon names
-    mon_endpoints=$(echo ${endpoints} | sed 's/[a-z]\+=//g')
+    mon_endpoints=$(echo ${endpoints} | sed 's/[a-z0-9_-]\+=//g')
```

Comment:
@vasukulkarni Can you elaborate on the scenario where you needed this? This is expected to translate from the mons in the configmap to the format that ceph recognizes. The format is a comma-separated list of `monName=endpoint` and we remove the monNames since those are a rook artifact. For example, in my cluster it would translate from:
```
a=10.96.19.219:6789,b=10.96.167.147:6789,c=10.96.251.180:6789
```
to
```
10.96.19.219:6789,10.96.167.147:6789,10.96.251.180:6789
```
The part before the `=` is just the rook mon name rather than the host name.

Reply:
@travisn I am seeing this in external cluster,  The ceph mons I get is: ceph-fedora-1591651937988-node3-monosd=10.0.102.118:6789,ceph-fedora-1591651937988-node2-monmdsosd=10.0.103.21:6789,ceph-fedora-1591651937988-node1-monmgrinstaller=10.0.103.43:6789
```
$ sudo ceph -s
  cluster:
    id:     647de7c2-e2c3-45aa-92a5-069165435828
    health: HEALTH_WARN
            clock skew detected on mon.ceph-fedora-1591651937988-node2-monmdsosd, mon.ceph-fedora-1591651937988-node1-monmgrinstaller
 
  services:
    mon: 3 daemons, quorum ceph-fedora-1591651937988-node3-monosd,ceph-fedora-1591651937988-node2-monmdsosd,ceph-fedora-1591651937988-node1-monmgrinstaller (age 4h)
    mgr: ceph-fedora-1591651937988-node1-monmgrinstaller(active, since 46h)
    mds: cephfs:1 {0=ceph-fedora-1591651937988-node2-monmdsosd=up:active}
    osd: 11 osds: 11 up (since 4h), 11 in (since 12d)
    rgw: 1 daemon active (ceph-fedora-1591651937988-node2-monmdsosd.rgw0)
 
  task status:
    scrub status:
        mds.ceph-fedora-1591651937988-node2-monmdsosd: idle
 
  data:
    pools:   12 pools, 768 pgs
    objects: 424 objects, 530 MiB
    usage:   16 GiB used, 47 GiB / 63 GiB avail
    pgs:     768 active+clean
```

Reply:
Got it, the external cluster endpoints won't follow the rook pattern. Were you able to test that change? It just looks like it would be too aggressive at removing characters, but maybe it will be fine. If it's working, could you just add a comment about the regex matching for future reference?

Reply:
@travisn I have tested this locally and it works fine

```
sh-4.2# cat /etc/ceph/ceph.conf 
[global]
mon_host = ceph-fedora-1591651937988-node3-10.0.102.118:6789,ceph-fedora-1591651937988-node2-10.0.103.21:6789,ceph-fedora-1591651937988-node1-10.0.103.43:6789

[client.admin]
keyring = /etc/ceph/keyring

sh-4.2# sh /usr/local/bin/toolbox.sh 
Tue Jul 14 17:11:09 UTC 2020 writing mon endpoints to /etc/ceph/ceph.conf: ceph-fedora-1591651937988-node3-monosd=10.0.102.118:6789,ceph-fedora-1591651937988-node2-monmdsosd=10.0.103.21:6789,ceph-fedora-1591651937988-node1-monmgrinstaller=10.0.103.43:6789
^C
sh-4.2# cat /etc/ceph/ceph.conf 
[global]
mon_host = 10.0.102.118:6789,10.0.103.21:6789,10.0.103.43:6789

[client.admin]
keyring = /etc/ceph/keyring
```

Reply:
@vasukulkarni Looks good. Could you just add a comment in the script about the different format for external clusters? thanks

Reply:
@travisn let me know if it looks good, I pushed new changes. Thanks!